### PR TITLE
Add module digital_ocean_droplet_info

### LINKED ISSUE #53371: digital_ocean_droplet should have a facts module
##### SUMMARY
Because the `digital_ocean` module is being deprecated, there is no way to query existing droplets. There should be a `digital_ocean_droplets_facts` module to glean information

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
digital_ocean

### PR DESCRIPTION
**Pull request by boutetnico**
##### SUMMARY
New module allowing to gather information about DigitalOcean droplets.

Fixes https://github.com/ansible/ansible/issues/53371

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
`digital_ocean_droplet_info`

##### ADDITIONAL INFORMATION
After reviewing other DO `_info` modules, I noticed they are not consistant in their return types. For example [digital_ocean_load_balancer_info](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/digital_ocean/digital_ocean_load_balancer_info.py) returns a `dict` if a `load_balancer_id` is provided, or returns a `list` otherwise.
In this PR, the return type is consistant: a `list`.
This is my first module and will he happy to get feedback and improve this PR.

### CODE REVIEW

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet_info.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    tag_name: "staging"
+'''
+
+
+RETURN = '''
+# Digital Ocean API info https://developers.digitalocean.com/documentation/v2/#list-all-droplets
+data:
+    description: List of droplets on DigitalOcean
+    returned: success and no resource constraint
+    type: dict
```
Comment (pmarques) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet_info.py]: As per your comment, should the type be `list` ?

**Review by pmarques — COMMENTED** [review]
Nice one @boutetnico , just one comment

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet_info.py (hunk 1 above)
Comment (boutetnico) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet_info.py]: 🤦‍♂Nice catch! Fixed.

**Review by pmarques — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
Hi @boutetnico, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.digitalocean<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**